src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -227,264 +207,420 @@ InferenceEngine::Precision Snippet::getRuntimePrecision() const {
-        size_t minimalJitWorkAmount = 256;
-        size_t currentJitWorkAmount = exec_domain.back();
+        size_t currentJitWorkAmount = domain[domain.size() - 1];
         while (currentJitWorkAmount < minimalJitWorkAmount && currentJitWorkAmount < fullWorkAmount) {
-            if (static_cast<int>(exec_domain.size()) - collapsedDims - 2 < 0)
+            if (static_cast<int>(domain.size()) - collapsedDims - 2 < 0)
                 break;
 
             bool canCollapse = true;
-            for (size_t i = 0; i < dims_in.size(); i++) {
-                if ((dims_in[i][dims_in[i].size() - 2] != 1 && dims_in[i][dims_in[i].size() - 1] == 1) ||
-                    (dims_in[i][dims_in[i].size() - 2] == 1 && dims_in[i][dims_in[i].size() - 1] != 1)) {
+            for (size_t i = 0; i < inputShapes.size(); i++) {
+                const size_t last = inputShapes[i].size() - 1;
+                if ((inputShapes[i][last - 1] != 1 && inputShapes[i][last] == 1) ||
+                    (inputShapes[i][last - 1] == 1 && inputShapes[i][last] != 1)) {
                     canCollapse = false;
                     break;
                 }
             }
 
-            size_t nextJitWorkAmount = currentJitWorkAmount * exec_domain[exec_domain.size() - 2];
+            size_t nextJitWorkAmount = currentJitWorkAmount * domain[domain.size() - 2];
             if (fullWorkAmount / nextJitWorkAmount >= minimalConcurrency) {
                 currentJitWorkAmount = nextJitWorkAmount;
                 // if we cannot use dim collapsing we should use tile2D
                 if (!canCollapse) {
-                    if (tileRank < maxTileRank) {
-                        tileRank++;
+                    if (TileRank < maxTileRank) {
+                        TileRank++;
                         continue;
                     }
 
                     break;
                 }
-
                 collapsedDims++;
-                for (auto &d : dims_in)
+                for (auto &d : inputShapes)
                     collapseLastDims(d, 1);
-
-                for (auto &d : dims_out)
+                for (auto &d : outputShapes)
                     collapseLastDims(d, 1);
-
-                collapseLastDims(exec_domain, 1);
+                collapseLastDims(domain, 1);
             } else {
                 break;
             }
         }
-        return collapsedDims;
+        return domain;
+    };
+    findDimsToCollapse();
+}
+ov::PartialShape Snippet::canonicalizeBody() {
+    auto edgeToBlockedShape = [](const EdgePtr& edge) {
+        const auto blockedDesc = edge->getMemory().GetDescWithType<BlockedMemoryDesc>();
+        std::vector<Dimension> dims;
+        // if blockDim == Shape::UNDEFINED_DIM, then it's a dynamic dimension, and we need to recreate a proper dynamic Dim
+        for (const auto& d : blockedDesc->getBlockDims())
+            dims.emplace_back(d == Shape::UNDEFINED_DIM ? 0 : d, d);
```

Comment:
I'm not sure that there should be `0` for `Shape::UNDEFINED_DIM` if you added `Subgraph::DYNAMIC_DIMENSION`. Could you please verify?